internal/api/create_credentials_v1alpha1.go

```diff
@@ -34,34 +41,141 @@ func (s *server) CreateCredentials(
 		return nil, connect.NewError(connect.CodeUnimplemented, errSecretManagementDisabled)
 	}
 
-	creds := credentials{
-		project:        req.Msg.GetProject(),
-		name:           req.Msg.GetName(),
-		description:    req.Msg.GetDescription(),
-		credType:       req.Msg.GetType(),
-		repoURL:        req.Msg.GetRepoUrl(),
-		repoURLIsRegex: req.Msg.GetRepoUrlIsRegex(),
-		username:       req.Msg.GetUsername(),
-		password:       req.Msg.GetPassword(),
+	credType := req.Msg.GetType()
+
+	if err := validateFieldNotEmpty("type", credType); err != nil {
+		return nil, err
+	}
+
+	switch credType {
+	case kargoapi.CredentialTypeLabelValueGit,
+		kargoapi.CredentialTypeLabelValueHelm,
+		kargoapi.CredentialTypeLabelValueImage:
+		creds := specificCredentials{
+			project:        req.Msg.GetProject(),
+			name:           req.Msg.GetName(),
+			description:    req.Msg.GetDescription(),
+			credType:       req.Msg.GetType(),
+			repoURL:        req.Msg.GetRepoUrl(),
+			repoURLIsRegex: req.Msg.GetRepoUrlIsRegex(),
+			username:       req.Msg.GetUsername(),
+			password:       req.Msg.GetPassword(),
+		}
+
+		kubernetesSecret, err := s.createSpecificCredentials(ctx, creds)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return connect.NewResponse(
+			&svcv1alpha1.CreateCredentialsResponse{
+				Credentials: kubernetesSecret,
+			},
+		), nil
+	case kargoapi.CredentialTypeLabelValueGeneric:
```

Comment:
I'm thinking a Secret that _isn't_ labeled as a generic credential should just automatically be treated as such. i.e. Perhaps generic credentials is the _default_ case?

The thing I think I want to avoid is Kargo insisting on a special label for Secret that _aren't_ really special in any way. (The "specific" creds _are_ special, so that doesn't bother me.)

Reply:
Or... I bet that label was basically necessary to make the existing credential-management endpoints work for generic Secrets? This actually has got me thinking about whether separate endpoints may be in order for Secrets since credentials are special in many ways that generic Secrets aren't.

For generic Secrets, I think we need the following:

* Get/list endpoints that redact _all_ values and annotations. (Compare to creds, which exempts the values of certain keys from being redacted.)

  Under the hood, credential and generic secret endpoints can share a common redaction func and pass the names of exempt keys.

* Update endpoint that updates/deletes individual keys. The generic update resource endpoint cannot be used for this purpose. Since we redact a lot on the get, the client doesn't _have_ everything it needs to send back for a full update.

  This is another bit that seems it could be shared between cred update and generic secret update endpoints.

* Delete endpoint.

* Create can use the generic create resource endpoint since I don't see any special treatment being required.